### PR TITLE
Add mechanism to remove idle workers

### PR DESCRIPTION
**Pull request by ljoss17**
Closes: #3385

## Description

______

### PR author checklist:
- [x] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [x] Added tests: integration (for Hermes) or unit/mock tests (for modules).
- [x] Linked to GitHub issue.
- [x] Updated code comments and documentation (e.g., `docs/`).
- [ ] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: crates/relayer/src/supervisor.rs [hunk 1]
```diff
@@ -280,6 +283,18 @@ pub fn spawn_rest_worker<Chain: ChainHandle>(
     )
 }
 
+/// Spawn a background task which verifies if there are idle workers and removes them if.
+pub fn spawn_cleanup_worker(workers: Arc<RwLock<WorkerMap>>) -> TaskHandle {
+    spawn_background_task(
+        error_span!("cleanup_worker"),
+        Some(Duration::from_secs(300)),
```
Comment (romac) [inline crates/relayer/src/supervisor.rs:290]: That sounds a bit excessive. Let's do it once per second instead.

File: crates/relayer/src/worker/handle.rs [hunk 2]
```diff
@@ -116,6 +116,18 @@ impl WorkerHandle {
         true
     }
 
+    /// Verify if at least one task of the WorkerHandle is stopped.
+    /// If it is the case, shutdown all remaining tasks.
+    pub fn is_stopped_then_kill(&self) -> bool {
```
Comment (romac) [inline crates/relayer/src/worker/handle.rs:121]:
```suggestion
    pub fn shutdown_stopped_tasks(&self) -> bool {
```

File: crates/relayer/src/worker/map.rs [hunk 3]
```diff
@@ -217,6 +217,21 @@ impl WorkerMap {
...
         }
     }
+
+    /// Verify if at least one task of the WorkerHandle is stopped.
+    /// If it is the case, shutdown all remaining tasks and remove the worker from the WorkerMap.
+    pub fn clean_stopped_workers(&mut self) {
+        let stopped_workers: Vec<(WorkerId, Object)> = self
+            .workers
+            .iter()
+            .filter(|w| w.1.is_stopped_then_kill())
```
Comment (romac) [inline crates/relayer/src/worker/map.rs:227]:
```suggestion
            .filter(|(_, w)| w.shutdown_stopped_tasks())
```

File: crates/relayer/src/worker/map.rs [hunk 4]
```diff
@@ -217,6 +217,21 @@ impl WorkerMap {
...
+            .map(|w| (w.1.id(), w.0.clone()))
```
Comment (romac) [inline crates/relayer/src/worker/map.rs:228]:
```suggestion
            .map(|(o, w)| (o.id(), w.clone()))
```

File: crates/relayer/src/worker/packet.rs [hunk 5]
```diff
@@ -40,6 +40,10 @@ use super::WorkerCmd;
 const INCENTIVIZED_CACHE_TTL: Duration = Duration::from_secs(10 * 60);
 const INCENTIVIZED_CACHE_MAX_CAPACITY: u64 = 1000;
 
+// Number of NewBlock consecutive NewBlock events before aborting the
+// packet cmd worker.
+const PACKET_CMD_WORKER_IDLE_TIMEOUT: u64 = 100;
```
Comment (romac) [inline crates/relayer/src/worker/packet.rs:45]:
```suggestion
const IDLE_TIMEOUT_BLOCKS: u64 = 100;
```

File: crates/relayer/src/supervisor.rs (hunk 1 above)
Comment (ljoss17) [reply crates/relayer/src/supervisor.rs:290]: Ok, this should also drastically reduce how long the test runs

File: .changelog/unreleased/improvements/ibc-relayer/10-cleanup-idle-workers.md [hunk 6]
```diff
@@ -0,0 +1,2 @@
+- Added a mechanism which removes idle workers
```
Comment (romac) [inline .changelog/unreleased/improvements/ibc-relayer/10-cleanup-idle-workers.md:1]:
```suggestion
- Terminate inactive workers to reduce the consumption of host system resources
```

**Pull request merged**